cluster-operator/src/main/java/io/strimzi/operator/cluster/model/Utils.java

```diff
@@ -0,0 +1,127 @@
+        try {
+            if (data.get(key) == null) { // value is not set in config map
+                return new Storage(Storage.StorageType.PERSISTENT_CLAIM);
+            }
+            return Storage.fromJson(new JsonObject(data.get(key)));
+        } catch (Exception e) {
+            throw new InvalidConfigMapException(key, " is corrupted");
+        }
+    }
+
+    public static String getNonemptyString(Map<String, String> data, String key, String defaultValue) {
+        if (data.get(key) == null) { // value is not set in config map -> will be used default value
+            return defaultValue;
+        }
+        if (data.get(key).length() == 0) {
+            throw new InvalidConfigMapException(key, " is empty");
+        }
+        return data.get(key);
+    }
+
+    public static String getJsonCorruptionBlame(DecodeException de, String config) {
+        String token = "";
+        token = de.getMessage();
+        if (token.contains("Illegal unquoted character")) {
+            return "JSON quotation";
+        }
+        token = token.replace("Failed to decode: Unrecognized token '", "");
+        token = token.substring(0, token.indexOf("'"));
+
+        String[] entries = config.replace("{", "").replace("}", "")
+               .replace("\"", "").replace(" ", "")
+               .replace("\n", "").split(",");
+        int i;
+        String blame = "";
+        for (i = 0; i < entries.length; ++i) {
+            if (entries[i].split(":")[1].equals(token)) {
+                blame = entries[i].split(":")[0];
+                break;
+            }
+        }
+        return blame;
+    }
+
+    public static JsonObject getConfig(Map<String, String> data, String key) {
+        String config = data.get(key);
+        if (config == null) {
+            return null;
+        }
+        try {
+            Map<String, Object> map = new JsonObject(config).getMap();
+            Map<String, String> numMap = new HashMap<String, String>();
+            Map<String, String> boolMap = new HashMap<String, String>();
+            Map<String, String> stringMap = new HashMap<String, String>();
+            for (Map.Entry<String, Object> entry : map.entrySet()) {
+                if (entry.getValue() instanceof String) {
+                    stringMap.put(entry.getKey(), (String) entry.getValue());
+                } else if (entry.getValue() instanceof Integer || entry.getValue() instanceof Long || entry.getValue() instanceof Double || entry.getValue() instanceof Float) {
+                    numMap.put(entry.getKey(), String.valueOf(entry.getValue()));
+                } else if (entry.getValue() instanceof Boolean) {
+                    boolMap.put(entry.getKey(), String.valueOf(entry.getValue()));
+                }  else {
+                    throw new InvalidConfigMapException(entry.getKey(), " has JSON with key " + entry.getKey() + " with a non-String value");
```

Comment:
We now allow even other values than just String to pass, so we should fix the error message in the Exception (i.e. the `non-string value` doesn't apply anymore).